packages/syft/src/syft/types/twin_object.py

```diff
@@ -30,35 +34,27 @@ class TwinObject(SyftObject):
 
     __attr_searchable__ = []
 
+    id: UID
     private_obj: ActionObject
-    private_obj_id: UID
+    private_obj_id: Optional[UID]
     mock_obj: ActionObject
-    mock_obj_id: UID
-
-    def __init__(
-        self,
-        private_obj: ActionObject,
-        mock_obj: ActionObject,
-        private_obj_id: Optional[UID] = None,
-        mock_obj_id: Optional[UID] = None,
-        id: Optional[UID] = None,
-    ) -> None:
-        private_obj = to_action_object(private_obj)
-        mock_obj = to_action_object(mock_obj)
-
-        if private_obj_id is None:
-            private_obj_id = private_obj.id
-        if mock_obj_id is None:
-            mock_obj_id = mock_obj.id
-        if id is None:
-            id = UID()
-        super().__init__(
-            private_obj=private_obj,
-            private_obj_id=private_obj_id,
-            mock_obj=mock_obj,
-            mock_obj_id=mock_obj_id,
-            id=id,
-        )
+    mock_obj_id: Optional[UID]
```

Comment:
I feel like here it's better to leave `private_obj_id: UID. # type: ignore` and `mock_obj_id: UID. # type: ignore`. Because of the pydantic validator these 2 are never gonna be None so the type is more accurate this way. Plus it's used in [pydantic doc](https://docs.pydantic.dev/latest/usage/validators/#validate-always) and there's an open [PEP](https://peps.python.org/pep-0712/) for this so I think this is ok we're just doing `# type: ignore` to work around the current type checker imperfection.